txinput.go

```diff
@@ -2,14 +2,25 @@ package bt
 
 import (
 	"bytes"
+	"context"
 	"encoding/binary"
+	"errors"
 	"fmt"
 
 	"github.com/libsv/go-bk/crypto"
 
 	"github.com/libsv/go-bt/v2/bscript"
 )
 
+// ErrNoInput signals the InputGetterFunc has reached the end of its input.
+var ErrNoInput = errors.New("no remaining inputs")
+
+// InputGetterFunc is used for FromInputs. It expects *bt.Input to be returned containing
+// relevant input information, and an err informing any retrieval errors.
+//
+// It is expected that bt.ErrNoInput will be returned once the input source is depleted.
+type InputGetterFunc func(ctx context.Context) (*Input, error)
```

Comment:
i see this went thought a few iterations haha but i still think it may be able to go for 1 more iteration to make it little clearer - let's discuss on monday 👍

Reply:
actually i have a much better idea of how i would like this to work and it follows the reasoning that we were discussing about potentially making inputs and outputs private that are used in a tx - since (unless someone can convince me otherwise) we will never have an input or output that isn't part of a tx and it just creates confusion, as we've seen previously, when separating them.

1st: we should probably return an array here instead of just one (input) since many times you will have an array of utxos to choose from and then we can add functionality to choose different utxos and stuff. plus of the utxogetter involved i/o then we'd want to minimise that and create a bulk call or something like that which we've done many times before.

2nd: it doesn't really feel accurate to say input getter since we're not getting the inputs, we're getting the utxos and creating the input... i think it would be much better to change this to UTXOGetterFunc and have it return an array of utxos and then we internally create the input from the utxo(s) and add them to the tx.

i'm still unsure whether we should still make the inputs/outputs private (ideally we could just leave them public to retain flexibility for users that know what they're doing but to remove all functions that imply separation between inputs/outputs and txs - for example the NewInput/Ouput funcs)... 

thoughts?

Reply:
- add satoshis as a param and return array instead of only 1 element
- create new data type: UTXO with the following 4 fields: prevTxID string, vout uint32, prevTxLockingScript string, satoshis uint64
- refactor tx.From() to take UTXO instead of 4 params
- rename InputGetterFunc to UTXOGetterFunc and return array of UTXOs